iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/DeviceEmulator.java

```diff
@@ -149,6 +135,22 @@ else if (client instanceof ModuleClient)
                     this.deviceMethodCallback, this.deviceMethodCallbackContext,
                     this.deviceMethodStatusCallback, this.deviceMethodStatusCallbackContext);
         }
+
+        long startTime = System.currentTimeMillis();
+        while (deviceStatus.statusOk == 0)
```

Comment:
Adding this step to ensure that the device methods subscription message gets acknowledged before continuing.